src/middlewares/upload_logo.ts

```diff
@@ -0,0 +1,111 @@
+/**
+ * Middleware that get tasks performed to upload a logo
+ */
+import ApiManagementClient from "azure-arm-apimanagement";
+import {
+  IResponseSuccessRedirectToResource,
+  ResponseErrorForbiddenNotAuthorized,
+  ResponseErrorInternal,
+  ResponseErrorNotFound,
+  ResponseSuccessRedirectToResource
+} from "italia-ts-commons/lib/responses";
+import { NonEmptyString } from "italia-ts-commons/lib/strings";
+import {
+  getApimUser,
+  IExtendedUserContract,
+  isAdminUser
+} from "../apim_operations";
+import { AdUser } from "../bearer_strategy";
+import * as config from "../config";
+
+import { Logo as ApiLogo } from "../../generated/api/Logo";
+
+import {
+  fromLeft,
+  fromPredicate,
+  taskEither,
+  TaskEither,
+  tryCatch
+} from "fp-ts/lib/TaskEither";
+
+import { ErrorResponses, notificationApiClient } from "../controllers/services";
+
+import { toError } from "fp-ts/lib/Either";
+
+export const getApimUserTask = (
+  apiClient: ApiManagementClient,
+  authenticatedUser: AdUser
+): TaskEither<ErrorResponses, IExtendedUserContract> =>
+  tryCatch(
+    () => getApimUser(apiClient, authenticatedUser.emails[0]),
+    errors => ResponseErrorInternal(toError(errors).message)
+  ).foldTaskEither<ErrorResponses, IExtendedUserContract>(
+    error => fromLeft(error),
+    response =>
+      response.isSome()
+        ? taskEither.of(response.value)
+        : fromLeft(
+            ResponseErrorNotFound(
+              "API user not found",
+              "Cannot find a user in the API management with the provided email address"
+            )
+          )
```

Comment:
Since `response` is an option we should name it as `maybeResponse`. Aside can we use `fold`?

Reply:
Thanks, 
`fold` is better.